app/src/unitTests/java/com/example/features/dashboard/presenter/MainPresenterTest.java

```diff
@@ -49,7 +53,7 @@
     @Before
     public void setUp() throws Exception {
         MockitoAnnotations.initMocks(this);
-        RxSchedulers rxSchedulers = new RxSchedulers(Schedulers.immediate(), Schedulers.immediate(), Schedulers.immediate(), Schedulers.immediate());
+        RxSchedulers rxSchedulers = new RxSchedulers(Schedulers.io(), Schedulers.computation(), Schedulers.trampoline(), AndroidSchedulers.mainThread());
```

Comment:
the RxSchedulers class acts as a wrapper that I can inject with the immediate() schedulers for tests. 
In rxjava 1 there's another way to handle this that looks like this:   
```
RxJavaHooks.setOnIOScheduler(scheduler -> Schedulers.immediate());
```
to change a specific Scheduler.

In rxjava 2 it's moved to the RxJavaPlugins class:

```
RxJavaPlugins.setIoSchedulerHandler(scheduler -> Schedulers.trampoline());
```

(note that immediate() scheduler no longer exists.)

Just pointing this out, you don't need to make this change, I'll do it in the very near future.

The only thing you need to do to make the tests pass it to change the line 

```
  RxSchedulers rxSchedulers = new RxSchedulers(Schedulers.immediate(), Schedulers.immediate(), Schedulers.immediate(), Schedulers.immediate());
```

to 

```
        RxSchedulers rxSchedulers = new RxSchedulers(Schedulers.trampoline(), Schedulers.trampoline(), Schedulers.trampoline(), Schedulers.trampoline());
```
and I'll sort out the rest in another PR